core/modules/node/node.module

```diff
@@ -1888,6 +1886,34 @@ function node_block_view($delta = '', $settings = array()) {
         }
       }
       break;
+
+    case 'empty':
+      // Check if the promoted view exists and still has the block display.
+      $view = views_get_view('promoted');
+      if ($view && $view->access('block_1')) {
+        $nids = views_get_view_result('promoted', 'block_1');
```

Comment:
Hm, I'd need to see if this will execute the query twice or not. I'm pretty sure every time you call `views_get_view()` it gives you a fresh copy of the view that hasn't had the result executed yet. You might be able to use `hook_views_pre_render()` instead to inject alternative empty text if none is otherwise specified.